libs/langchain/langchain/runnables/openai_functions.py

```diff
@@ -0,0 +1,46 @@
+from operator import itemgetter
+from typing import Any, Callable, List, Mapping, Optional, Union
+
+from typing_extensions import TypedDict
+
+from langchain.output_parsers.openai_functions import JsonOutputFunctionsParser
+from langchain.schema.output import ChatGeneration
+from langchain.schema.runnable import RouterRunnable, Runnable, RunnableBinding
+
+
+class OpenAIFunction(TypedDict):
+    """A function description for ChatOpenAI"""
+
+    name: str
+    """The name of the function."""
+    description: str
+    """The description of the function."""
+    parameters: dict
+    """The parameters to the function."""
+
+
+class OpenAIFunctionsRouter(RunnableBinding[ChatGeneration, Any]):
+    """A runnable that routes to the selected function."""
+
+    functions: Optional[List[OpenAIFunction]]
```

Comment:
* why accept non empty collections or Nones?  Is the idea to be able to disable tool calling without having to change the structure of the chain? 
* does empty list work?